src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -69,6 +71,59 @@ internal bool TryGetNextHeader(Stream archiveStream, bool copyData)
             }
         }
 
+        // Asynchronously attempts read all the fields of the next header.
+        // Throws if end of stream is reached or if any data type conversion fails.
+        // Returns true if all the attributes were read successfully, false otherwise.
+        internal async ValueTask<bool> TryGetNextHeaderAsync(Stream archiveStream, bool copyData, CancellationToken cancellationToken)
+        {
+            // The four supported formats have a header that fits in the default record size
+            IMemoryOwner<byte> rented = MemoryPool<byte>.Shared.Rent(minBufferSize: TarHelpers.RecordSize);
```

Comment:
This should use ArrayPool directly rather than MemoryPool.

Reply:
But a couple of lines below I need to pass the `Memory<byte>` to `archiveStream.ReadExactlyAsync`.

Reply:
Sure... but arrays are implicitly `Memory<byte>` (or if you want to slice it, then explicitly use AsMemory)

```
byte[] rented = ArrayPool<byte>.Shared.Rent(TarHelpers.RecordSize);
Memory<byte> buffer = rented.AsMemory(0, TarHelpers.RecordSize);

...

ArrayPool<byte>.Shared.Return(rented);
```

Reply:
TIL. Thanks.

Reply:
> Sure... but arrays are implicitly `Memory<byte>` (or if you want to slice it, then explicitly use AsMemory)

Thanks for this explanation in advance. I am starting to see some test failures when attempting to do async writes using the existing code that takes spans, like this:

https://github.com/dotnet/runtime/blob/70e9ca05739971856f17dfd23f44bd72e153cc2d/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs#L207-L212